CodeEdit/Features/Tasks/Views/StopToolbarButton.swift

```diff
@@ -0,0 +1,94 @@
+    /// changes.
+    @State private var statusListener: AnyCancellable?
+
+    var body: some View {
+        HStack {
+            if let currentSelectedStatus, currentSelectedStatus == .running {
+                    Button {
+                        taskManager.terminateActiveTask()
+                    } label: {
+                        Label("Stop", systemImage: "stop.fill")
+                            .labelStyle(.iconOnly)
+                            .font(.system(size: 15, weight: .regular))
+                            .help("Stop selected task")
+                            .frame(width: 28)
+                            .offset(y: 1.5)
+                    }
+                    .frame(height: 22)
+                    .transition(.opacity.combined(with: .move(edge: .trailing)))
+            }
+        }
+        .frame(width: 38, height: 22)
+        .animation(
+            .easeInOut(duration: 0.3),
+            value: currentSelectedStatus
+        )
+        .onChange(of: taskManager.selectedTaskID) { _ in updateStatusListener() }
+        .onChange(of: taskManager.activeTasks) { _ in updateStatusListener() }
+        .onAppear(perform: updateStatusListener)
+        .onDisappear {
+            statusListener?.cancel()
+        }
+    }
+
+    /// Update the ``statusListener`` to listen to a potentially new active task.
+    private func updateStatusListener() {
+        statusListener?.cancel()
+        currentSelectedStatus = taskManager.activeTasks[taskManager.selectedTaskID ?? UUID()]?.status
+        guard let id = taskManager.selectedTaskID else { return }
+        statusListener = taskManager.activeTasks[id]?.$status.sink { newValue in
+            currentSelectedStatus = newValue
+        }
+    }
+}
+
+struct StartTaskToolbarButton: View {
+    @UpdatingWindowController var windowController: CodeEditWindowController?
+
+    // TODO: try to get this from the environment
+    @ObservedObject var taskManager: TaskManager
+    @EnvironmentObject var workspace: WorkspaceDocument
+
+    var utilityAreaCollapsed: Bool {
+        windowController?.workspace?.utilityAreaModel?.isCollapsed ?? true
+    }
+
+    var body: some View {
+        Button {
+            self.runActiveTask()
+            if utilityAreaCollapsed {
+                CommandManager.shared.executeCommand("open.drawer")
+            }
+            workspace.utilityAreaModel?.selectedTab = .debugConsole
+            taskManager.taskShowingOutput = taskManager.selectedTaskID
+        } label: {
+            Label("Start", systemImage: "play.fill")
+                .labelStyle(.iconOnly)
+                .font(.system(size: 18, weight: .regular))
+                .help("Start selected task")
+                .frame(width: 28)
+                .offset(CGSize(width: 0, height: 2.5))
+        }
+    }
+
+    private func runActiveTask() {
+        taskManager.executeActiveTask()
+    }
```

Comment:
That seems a bit unnecessary.

Reply:
It is, brought it over from CodeEditWindowController+Toolbar